### PR TITLE
Initial Discord documentation

### PR DESCRIPTION
**Pull request by valgrimm**
This documentation does not describe how to set up channels, but is otherwise probably sufficient to get newcomers started.
I'd like to add an animated GIF to make accessing the Notifications menu clearer, but cannot upload files. right now there is a placeholder -IMAGE GOES HERE-.

### CODE REVIEW

File: src/chat/discord.md [hunk 1]
```diff
@@ -1,4 +1,104 @@
 ## Discord
 
-We haven't yet written the documentation for Discord, but if you'd like to help
-out, this is the place for it to go!
+Rust's Discord is used by Community, Lang, Ops, and Documentation teams, as well as their working groups. It is also the home of Domain Working Groups and provides a space for general discussion among Rust users, contributors, and beginners.
```
Comment (Centril) [inline src/chat/discord.md]: Lang team has moved to Zulip.

Comment (khionu) [reply src/chat/discord.md]: Phrasing should be oriented around the Discord being *maintained* for Rust collaboration, not that it is their fixed home.

File: src/chat/discord.md [hunk 2]
```diff
@@ -1,4 +1,104 @@
...
+# Where to go for help with using Discord
+Discord's support center provides documentation about its [user interface](https://support.discordapp.com/hc/en-us/categories/200404398) and [account settings](https://support.discordapp.com/hc/en-us/categories/200404358).
+
+# Getting started
+
+1) Understand community standards <br/>
+Before joining the conversation on Discord, you can prepare by reading the [Code of Conduct and Moderation Guidelines](https://www.rust-lang.org/community#conduct).
+
+2) Access channels <br/>
+To access the [Rust Discord channel](https://discordapp.com/invite/rust-lang) visit [(https://discordapp.com/invite/rust-lang)](https://discordapp.com/invite/rust-lang) and open the channel in your browser or app. If you do not already have a Discord account, you can register for one as part of the process of gaining access.
```
Comment (khionu) [inline src/chat/discord.md]: Can you use `https://discord.gg/rust-lang` here? It resolves to the former, but it's more SEO friendly. Also, I think only one link would be best, so people don't feel the need to check two links.

Would also be prudent to mention that they need to agree to our CoC by following the instructions in #welcome.

File: src/chat/discord.md [hunk 3]
```diff
@@ -1,4 +1,104 @@
...
+
+3) Configure notifications <br/>
+A good first step is disabling notifications for channels that are not yet relevant to you, so that you will not be overwhelmed with messages. 
+-IMAGE GOES HERE-
+Select the expansion arrow next to the server name banner (titled "The Rust Programming Language") and select Notifications from the dropdown. Then follow the configuration instructions provided [on the Discord Support site](https://support.discordapp.com/hc/en-us/articles/215253258-Notifications-Settings-101).
+
+# Appropriate conversation
+Discussions should be related to the channel purpose. On team channels, conversation should be related to team business.
+All channels are expected to be used for purposes related to the Rust project. Discussion of (for example) wildlife or sightseeing are not appropriate. Discord, like all official Rust spaces, is governed by the [Code of Conduct](https://www.rust-lang.org/community#conduct).
```
Comment (khionu) [inline src/chat/discord.md]: Given it'd already be mentioned in my suggested edit, as well in step 1, I'd remove the last sentence.

File: src/chat/discord.md [hunk 4]
```diff
@@ -1,4 +1,104 @@
...
+
+# Channels
+
+The following channels are especially relevant to newcomers to the Rust project: 
+* rustlings - If you are learning Rust using the [Rustlings tutorials](https://github.com/rust-lang/rustlings), this channel is a great place to meet your fellow students.
```
Comment (khionu) [inline src/chat/discord.md]: #rustlings is meant for contributors to Rustlings, not for people who are new to Rust. I would specify that channels outside of General are for contributors to Rust, not consumers.

File: src/chat/discord.md [hunk 5]
```diff
@@ -1,4 +1,104 @@
...
+* wg-governance
+* wg-gamedev
+
+DOCUMENTATION TEAM
+* docs
+* rustlings
+* reference
+
+# Messages
+Discord is used asynchronously as well as synchronously, so you should not generally expect that your messages will receive a response quickly unless a meeting is taking place. Depending on how your notifications are configured, you will see a red circle on top of the Discord icon in your system tray when new messages are received. If you wish to communicate with a specific individual, right-click on their user icon and select "Message" in the dropdown menu.
```
Comment (khionu) [inline src/chat/discord.md]: I'd use less technical phrasing here. "Discord activity depends on when people are available, so..."

File: src/chat/discord.md [hunk 6]
```diff
@@ -1,4 +1,104 @@
...
-out, this is the place for it to go!
+Rust's Discord is used by Community, Lang, Ops, and Documentation teams, as well as their working groups. It is also the home of Domain Working Groups and provides a space for general discussion among Rust users, contributors, and beginners. 
+
+# Where to go for help with using Discord
+Discord's support center provides documentation about its [user interface](https://support.discordapp.com/hc/en-us/categories/200404398) and [account settings](https://support.discordapp.com/hc/en-us/categories/200404358).
+
+# Getting started
+
+1) Understand community standards <br/>
+Before joining the conversation on Discord, you can prepare by reading the [Code of Conduct and Moderation Guidelines](https://www.rust-lang.org/community#conduct).
```
Comment (khionu) [inline src/chat/discord.md]: Should add Discord's Community Guidelines as well. https://discordapp.com/guidelines

File: src/chat/discord.md [hunk 7]
```diff
@@ -1,4 +1,104 @@
...
+* contribute - Interested in contributing to the Rust project? In addition to joining the discussion on this channel, you can subscribe to the newsletter and learn more about project teams.  
+    * Newsletter <br/>
+Contribution opportunities are regularly posted in the [This Week In Rust](https://this-week-in-rust.org/) newsletter. 
+    * Team descriptions <br/>
+Find out more about [specific teams](https://www.rust-lang.org/governance#teams), including those with channels included in the list below.
+
+*Channel list*
+
+GENERAL
+* rust-usage
```
Comment (khionu) [inline src/chat/discord.md]: I would put #welcome here. It disappears after a person agrees to the CoC, but it's still there.

File: src/chat/discord.md [hunk 8]
```diff
@@ -1,4 +1,104 @@
...
+* rustlings - If you are learning Rust using the [Rustlings tutorials](https://github.com/rust-lang/rustlings), this channel is a great place to meet your fellow students. 
+* rust-usage - This is a channel where you can access support for resolving specific language use questions. The [Rust Users Forum](https://users.rust-lang.org/) is also relevant to your needs.
+* beginners - Here, you can meet people who began using Rust relatively recently. 
+* contribute - Interested in contributing to the Rust project? In addition to joining the discussion on this channel, you can subscribe to the newsletter and learn more about project teams.  
+    * Newsletter <br/>
+Contribution opportunities are regularly posted in the [This Week In Rust](https://this-week-in-rust.org/) newsletter. 
+    * Team descriptions <br/>
+Find out more about [specific teams](https://www.rust-lang.org/governance#teams), including those with channels included in the list below.
+
+*Channel list*
```
Comment (pietroalbini) [inline src/chat/discord.md]: I wouldn't include a list of channels here, we'll surely forget updating it.

File: src/chat/discord.md (hunk 2 above)
Comment (technetos) [reply src/chat/discord.md]: If they are new to the discord they will have to agree to the CoC in the #welcome channel

File: src/chat/discord.md (hunk 8 above)
Comment (technetos) [reply src/chat/discord.md]: I second this.  Im happy to maintain a list of general channels for general stuff but the team channels change too often.

Comment (khionu) [reply src/chat/discord.md]: I'm behind this as well

File: src/chat/discord.md (hunk 1 above)
Comment (valgrimm) [reply src/chat/discord.md]: Revised to address these comments. Thanks!

File: src/chat/discord.md (hunk 3 above)
Comment (valgrimm) [reply src/chat/discord.md]: Revised to address this comment. Thanks!

File: src/chat/discord.md (hunk 5 above)
Comment (valgrimm) [reply src/chat/discord.md]: Revised to address this comment. Thanks!

File: src/chat/discord.md (hunk 7 above)
Comment (valgrimm) [reply src/chat/discord.md]: Revised to address this comment. Thanks!

File: src/chat/discord.md (hunk 8 above)
Comment (valgrimm) [reply src/chat/discord.md]: Revised to address this comment. Thanks!

**Comment by valgrimm** [comment]
Ok, this should be good to go now, seems.

**Pull request merged**